gnuradio-runtime/lib/top_block_impl.cc

```diff
@@ -129,6 +132,21 @@ namespace gr {
   void
   top_block_impl::wait()
   {
+    do {
```

Comment:
What I was really asking was the need for do {} while(false) as an outer envelope for the rest of the code.